mne/annotations.py

```diff
@@ -331,3 +331,140 @@ def _read_annotations(fid, tree):
         annotations = Annotations(onset, duration, description,
                                   orig_time)
     return annotations
+
+
+@verbose
+def events_from_annotations(raw, annotations=None, event_id=None, regexp=None,
+                            event_id_func=None,
+                            convention="default", verbose=None):
+    """Get events and event_id from an Annotations object
+
+    Parameters
+    ----------
+    raw : instance of Raw
+        The raw data for which Annotations are defined.
+    annotations : instance of Annotations | None
+        The annotations. If None, raw.annotations will be used.
+    event_id : dict | None
+        Dictionary of string keys and integer values as used in mne.Epochs
+        to map annotation descriptions to integer event codes. If None,
+        all descriptions of annotations are mapped and assigned arbitrary
+        unique integer values. Else, only the keys present will be mapped
+        and and the annotations with other descriptions will be ignored.
+    regexp : str | None
+        Regular expression used to filter the annotations whose
+        descriptions is a match.
+    event_id_func : None | str | callable
+        What to do for events not found in ``event_id``. Must take one ``str``
+        argument and return an ``int``. If string, must be 'strip-to-integer',
+        in which case it defaults to stripping event codes such as "D128" or
+        "S  1" of their non-integer parts and returns the integer.
+        If the event is not in the ``event_id`` and calling ``event_id_func``
+        on it results in a ``TypeError`` (e.g. if ``event_id_func`` is
+        ``None``) or a ``ValueError``, the event is dropped.
+    convention : str
+        Can be "eeglab" or "default". XXX
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see
+        :func:`mne.verbose` and :ref:`Logging documentation <tut_logging>`
+        for more). Defaults to self.verbose.
+
+    Returns
+    -------
+    events : ndarray, shape (n_events, 3)
+        The events.
+    event_id : dict
+        The event_id variable that can be passed to Epochs.
+    """
+    import re
+
+    assert convention in ['default', 'eeglab']
+
+    if annotations is None:
+        annotations = raw.annotations
+
+    if convention == "default":
+        inds = raw.time_as_index(
+            annotations.onset, use_rounding=True) + raw.first_samp
```

Comment:
if annotations are still None (empty annotations field) this will result in AttributeError